### PR TITLE
Add support for `--type=secret` in `docker inspect`

### PR DESCRIPTION
**Pull request by vdemeester**
Taking into account @allencloud, add support for `docker inspect --type=secret` because, why not 👼.

/cc @icecrime :trollface: @thaJeztah @cpuguy83 @AkihiroSuda @dnephin @tiborvass @vieux 

🦁

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

**Comment by thaJeztah** [comment]
design LGTM

File: cli/command/system/inspect.go [hunk 1]
```diff
@@ -144,6 +149,10 @@ func inspectAll(ctx context.Context, dockerCli *command.DockerCli, getSize bool,
 			objectType:      "plugin",
 			objectInspector: inspectPlugin(ctx, dockerCli),
 		},
+		{
+			objectType:      "secret",
+			objectInspector: inspectSecret(ctx, dockerCli),
```
Comment (thaJeztah) [inline cli/command/system/inspect.go]: isSwarmObject: true,

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: cli/command/system/inspect.go (hunk 1 above)
Comment (vdemeester) [inline cli/command/system/inspect.go]: @thaJeztah nice I missed that one 👼

**Comment by allencloud** [comment]
That is great. Thanks a lot. @vdemeester 
Eh... Please do not forget to update inspect.md under command docs directory. 😄

**Comment by vdemeester** [comment]
>  Eh... Please do not forget to update inspect.md under command docs directory. 😄

Right, I was waiting for it to pass `design` phase but.. it's already the case 😛

**Comment by ehazlett** [comment]
design lgtm

**Comment by vdemeester** [comment]
@thaJeztah completions scripts updated (cc @sdurrheimer @albers)

**Review by thaJeztah — APPROVED** [review]
code LGTM, but needs updates to the documentation as mentioned above :smile:

**Comment by thaJeztah** [comment]
docs is being taken care of in https://github.com/docker/docker/pull/32112

**Comment by albers** [comment]
Bash completion tested locally, perfect!

**Comment by ehazlett** [comment]
LGTM

**Comment by thaJeztah** [comment]
pssst @ehazlett; looks like it can be moved to `status/4-merge` 😇

File: contrib/completion/zsh/_docker [hunk 2]
```diff
@@ -2713,6 +2713,8 @@ __docker_subcommand() {
                         __docker_complete_nodes && ret=0
                     elif [[ ${words[(r)--type=plugin]} == --type=plugin ]]; then
                         __docker_complete_plugins && ret=0
+                    elif [[ ${words[(r)--type=service]} == --type=secrets ]]; then
```
Comment (sdurrheimer) [inline contrib/completion/zsh/_docker]: Sorry for looking at this too late, should be `elif [[ ${words[(r)--type=secrets]}` not `elif [[ ${words[(r)--type=service]}`

Comment (vdemeester) [inline contrib/completion/zsh/_docker]: oh shit, my bad I missed one 😱 I'll fix 👼

Comment (thaJeztah) [inline contrib/completion/zsh/_docker]: Oh, thanks of the catch; can you open a pull request to fix that, or want someone else to do that?

**Pull request merged**